framework/tlvf/AutoGenerated/include/tlvf/CmduMessage.h

```diff
@@ -62,6 +62,15 @@ class CmduMessage {
         return std::dynamic_pointer_cast<T>(getClass(idx));
     }
 
+    template <class T> std::shared_ptr<T> getClass() const
+    {
+        for (size_t idx = 0; idx < getClassCount(); idx++) {
+            if (auto c = getClass<T>(idx))
+                return c;
+        }
+        return nullptr;
+    }
```

Comment:
You are missing something probably, I'll try to explain:
`getClass<T>(idx)` is actually `std::dynamic_pointer_cast<T>(getClass(idx))`.
So, this basically goes through each class in the vector and tries to dynamically cast it to the requested type.

Reply:
Then why the `return? it will return in the first iteration, no?
Still, I don't see what benefit it gives to cast it.

Reply:
> It will return in the first iteration, no?

No - `if (auto c = getClass<T>(idx))` will be `true` only if `getClass<T>(idx)` returned none NULL - which means the class at index `idx` is of type `T`.

> Also, I didn't see any use of it.

This is because this PR prepares all that's needed for automatic parsing but doesn't change any logic outside of TLVF to do the automatic parsing. It is used in `tlvf_test`:

```
CmduMessageRx received_message;
received_message.parse(recv_buffer, sizeof(recv_buffer), true, true);
auto tlv1_ = received_message.getClass<tlvNon1905neighborDeviceList>();
  if (!tlv1_)
    errors++;
```

As you can see - this function allows the user to simply call getClass<TYPE> and get nullptr if this TLV was not part of the CMDU, otherwise it returns the class it self already casted.

**Bottom line - this is one of the most important changes in this PR.**

Reply:
Thanks for the clarification! I can now understand that the loop will continue to iterate until the class in index `idx` will match to the function template T.
It wasn't so clear at the first look, so thanks.
Ok but what you do if you have several tlvs from the same type?
For example Channel Selection Response message may contain One or *more* Channel Selection Response TLVs.

Reply:
Most of the time you will just look for one class, so always returning a list is an overkill.
Moreover - The user of this API knows in advance if he has more than one element, in this case we will provide an additional API to return the class of type T at index idx.
We currently have such an API but it just tries to cast the class at idx `idx` out of *all classes* into type `T`. I will change it so that it will iterate through all classes, and only those which are of type `T` will be counted for - so it will behave as if there is an array which contains all the classes of type `T` and return the one at index `idx`. See last fixup for this.